usaspending_api/database_scripts/job_archive/reset_awards_subaward_rollups.py

```diff
@@ -0,0 +1,252 @@
+"""
+Jira Ticket Number(s): DEV-2116
+
+    DEV-2116 is about integrating Broker's new subaward table into USAspending.
+
+Expected CLI:
+
+    $ python usaspending_api/database_scripts/job_archive/reset_awards_subaward_rollups.py
+
+Purpose:
+
+    As part of DEV-2116, we will need to fully reload the subaward table.  It
+    was noticed during testing that awards without a subaward have inconsistent
+    "default" values - some are NULL and others are 0.  The default is supposed
+    to be NULL for the dollar figure and 0 for the count, based on the model
+    definition.
+
+    This script will run through all of the awards and fix the values for awards
+    that have no subawards.  This will prevent the actual subaward rebuild from
+    taking an immense amount of time, potentially locking tables for hours on end.
+
+    The single SQL script version took 5 hours-ish on DEV.  We're hoping this
+    script will pare that number down a bit and, since we're chunking/batching,
+    will not hold locks for extended periods of time.
+
+    This script can be eliminated after sprint 85 has finished rolling out and
+    we are happy with the results.
+
+"""
+import math
+import psycopg2
+import time
+
+from datetime import timedelta
+from os import environ
+
+
+# DEFINE THIS ENVIRONMENT VARIABLE BEFORE RUNNING!
+# Obvs, this is the connection string to the database.
+CONNECTION_STRING = environ["DATABASE_URL"]
+
+
+CHUNK_SIZE = 500000
+
+
+CREATE_TEMP_SUBAWARD_AWARD_ID_TABLE_SQL = """
+    drop table if exists    temp_dev_2006_subaward_award_ids;
+
+    create unlogged table   temp_dev_2006_subaward_award_ids
+    as select distinct      award_id
+    from                    subaward 
+    where                   award_id is not null;
+"""
+
+
+INDEX_TEMP_TABLE_SQL = """
+    create unique index     idx_temp_dev_2006_subaward_award_ids
+    on                      temp_dev_2006_subaward_award_ids (award_id) 
+"""
+
+
+GET_MIN_MAX_SQL = "select min(id), max(id) from awards"
+
+
+# Reset award subaward rollups to their defaults for awards that do not have
+# subawards and where they are not already the correct value.
+UPDATE_AWARDS_CHUNK_SQL = """
+    update
+        awards
+
+    set
+        total_subaward_amount = null,
+        subaward_count = 0
+
+    where
+        id between {minid} and {maxid} and
+        id not in (
+            select award_id from temp_dev_2006_subaward_award_ids where award_id between {minid} and {maxid}
+        ) and 
+        (
+            total_subaward_amount is not null or
+            subaward_count is distinct from 0
+        )
+"""
+
+
+DROP_TEMP_TABLE_SQL = "drop table if exists temp_dev_2006_subaward_award_ids"
+
+
+class Timer:
+    """
+    Stolen from usaspending_api/common/helpers/timing_helpers.Timer which,
+    unfortunately, we can't use since this is a standalone (throwaway) script.
+    """
```

Comment:
I really don't care for reviewing purposes, but why can't you use a library for a throwaway script?

Reply:
Two reasons.  1)  I like to think of throwaway scripts as all totally encapsulated since it acts as a record of something we did.  I don't want code it depends on to change.  Think of it like a snapshot in time.  2)  It's kind of a pain to import functions contained in files that are external to the current directory.  It's not worth the trouble for a throwaway script, IMO.